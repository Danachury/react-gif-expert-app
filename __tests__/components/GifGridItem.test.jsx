import { render, screen } from '@testing-library/react'
import GifGridItem from '../../src/components/GifGridItem.jsx'

describe('Tests in <GifGridItem />', () => {

    const title = 'Saitama'
    const url = 'https://one-punch-man/saitama.jpg'

    test('should match with snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url}/>)
        expect(container).toMatchSnapshot()
    })

    test('should show the image with the sent url and its alt', () => {
        render(<GifGridItem title={title} url={url}/>)
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('should show the Gif title', () => {
        render(<GifGridItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy()
    })
})
