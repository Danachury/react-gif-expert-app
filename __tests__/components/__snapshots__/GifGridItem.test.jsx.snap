// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch-man/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
